lib/core/types/index.ts

```diff
@@ -0,0 +1,69 @@
+import type ForkTsCheckerWebpackPlugin from 'fork-ts-checker-webpack-plugin';
+import type LoaderOptions from 'react-docgen-typescript-loader/dist/LoaderOptions';
+import { Configuration } from 'webpack';
+
+/**
+ * The interface for Storybook configuration in `main.ts` files.
+ */
+export interface StorybookConfig {
+  /**
+   * Sets the addons you want to use with Storybook.
+   *
+   * @example `['@storybook/addon-essentials']`
+   */
+  addons?: string[];
+  /**
+   * Tells Storybook where to find stories.
+   *
+   * @example `['./src/*.stories.(j|t)sx?']`
+   */
+  stories: string[];
```

Comment:
@ndelangen stories can be a function returning an array of strings right?

Reply:
If that's possible, we can add that @ndelangen @shilman :)

I'm hoping this is temporary @matheo, as this would be exported differently once we convert core to TS.

Reply:
All preset properties can be a `T | (T, Options) => T | (T, Options) => Promise<T>`

Reply:
@ndelangen That applies to `addons` and `stories`?

Specifically we're looking at the options here: https://github.com/storybookjs/storybook/blob/next/lib/core/types/index.ts#L8-L32

Reply:
`addons` & `presets` are the exception.
they are: `string[]`.

everything else is `T | (T, Options) => T | (T, Options) => Promise<T>`

Reply:
the technical reason is, as we "discover" presets/addons, we don't know the whole list yet, and so we can't pass it in. I figured it'd be too complex and confusing for presets to receive an incomplete list of presets.